lib/health_check/custom_health_checks.rb

```diff
@@ -0,0 +1,53 @@
+module HealthCheck
+  class CustomHealthChecks
+
+    def self.check_sidekiq_redis
+      Sidekiq.redis { |r| "" if r.ping == "PONG" }
+    rescue Exception => e
+       create_error 'sidekiq-redis', e.message
+    end
+
+    def self.check_redis
+      "" if Redis.new.ping
+    rescue Exception => e
+       create_error 'redis', e.message
+    end
+
+    def self.check_s3
+      return create_error 's3', "Could not connect to aws" if aws_s3_client.nil?
+      HealthCheck.buckets.each do |bucket|
+        aws_s3_client.put_object(bucket: bucket, key: 'FOO', body: 'BAR')
```

Comment:
Btw, not sure if it makes sense to use it here but we already have a gateway class in the `ah` gem, which abstracts the `put_object` and `get_object` S3 operations: https://github.com/AirHelp/ah/blob/master/lib/ah/gateways/s3.rb. We are now trying to use it across apps. But I'm not sure if it suits your app here.